dev/Cargo.toml

```diff
@@ -30,9 +30,12 @@ version = "0.0.1"
 [dependencies]
 anyhow = "1.0.95"
 clap = { version = "4.5.23", features = ["derive"] }
+enquote = "1.1.0"
 env_logger = "0.11.6"
+itertools = "0.13.0"
 log = "0.4.22"
-syn = { version = "2.0.91", features = ["visit","full","extra-traits"] }
+proc-macro2 = { version = "1.0.91", features = ["span-locations"] }
```

Comment:
It's better to avoid using `proc-macro2` directly. Manipulating the AST through spans doesn't make sense to me.

Reply:
I thought we had everything in https://docs.rs/syn/2.0.92/syn/struct.Attribute.html.

Reply:
> Yes, which and it's only a public api when we enable `proc-macro2/span-locations`

I'm confused. It's enabled under `full` or `derive`, which have already been enabled. Why is it related to another crate, `proc-macro2`?

Reply:
Hi, would you allow me to push changes directly to this PR?

Reply:
Got it. `MetaList` exposes `TokenStream` directly we have to use `proc_macro2`'s API to operate on it.

Thank you for this@!

Reply:
We can use `Attribute::parse_nested_meta` for this. 

Please note that `dev` is used by every developer and CI. We should be very careful to not include new deps in.